src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/controllers/CcdCallbackController.java

```diff
@@ -1,42 +1,52 @@
 package uk.gov.hmcts.reform.bulkscan.orchestrator.controllers;
 
+import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RestController;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.CallbackProcessor;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd.AttachCaseCallbackService;
 import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
 import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
 
+import java.util.List;
+
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 import static org.springframework.http.ResponseEntity.ok;
 
 @RestController
 public class CcdCallbackController {
 
-    private final CallbackProcessor callbackProcessor;
+    private final AttachCaseCallbackService attachCaseCallbackService;
 
     @Autowired
-    public CcdCallbackController(CallbackProcessor callbackProcessor) {
-        this.callbackProcessor = callbackProcessor;
+    public CcdCallbackController(AttachCaseCallbackService attachCaseCallbackService) {
+        this.attachCaseCallbackService = attachCaseCallbackService;
     }
 
     @PostMapping(
-        path = "/callback/{type}",
+        path = "/callback/attach_case",
         consumes = APPLICATION_JSON_VALUE,
         produces = APPLICATION_JSON_VALUE
     )
-    public ResponseEntity<CallbackResponse> handleCallback(
-        @PathVariable("type") String type,
-        @RequestBody CallbackRequest callback) {
+    public ResponseEntity<CallbackResponse> handleCallback(@RequestBody CallbackRequest callback) {
+        if (callback != null && callback.getCaseDetails() != null) {
+            return respondWith(attachCaseCallbackService.process(callback.getCaseDetails()));
+        } else {
+            return respondWith(ImmutableList.of("Internal Error: callback or case details were empty"));
```

Comment:
More like client error.
Also, it would be nice to add a comment explaining why we return `200 OK` here despite errors.

Reply:
hmmm ...  Ill refactor

Reply:
not agreeing nor disagreeing. but i'm in favour of @rkondratowicz. the way response is built it would be better to respond OK but with errors. it is client error - somebody called with empty sections in body therefore making a 400. but again, the structure of response suggests an alternative